closed/src/java.base/share/classes/sun/security/ec/NativeECDSASignature.java

```
@@ -0,0 +1,742 @@
+        }
+    }
+
+    private ECDSASignature getJavaInstance() {
+        if (this.messageDigest == null) {
+            return (this.p1363Format? new ECDSASignature.RawinP1363Format() :
+                new ECDSASignature.Raw());
+        } else {
+            String mdAlgo = messageDigest.getAlgorithm();
+            switch(mdAlgo) {
+                case "SHA1":
+                    return (this.p1363Format? new ECDSASignature.SHA1inP1363Format() :
+                        new ECDSASignature.SHA1());
+                case "SHA-224":
+                    return (this.p1363Format? new ECDSASignature.SHA224inP1363Format() :
+                        new ECDSASignature.SHA224());
+                case "SHA-256":
+                    return (this.p1363Format? new ECDSASignature.SHA256inP1363Format() :
+                    new ECDSASignature.SHA256());
+                case "SHA-384":
+                    return (this.p1363Format? new ECDSASignature.SHA384inP1363Format() :
+                    new ECDSASignature.SHA384());
+                case "SHA-512":
+                    return (this.p1363Format? new ECDSASignature.SHA512inP1363Format() :
+                    new ECDSASignature.SHA512());
+                case "SHA3-224":
+                    return (this.p1363Format? new ECDSASignature.SHA3_224inP1363Format() :
+                    new ECDSASignature.SHA3_224());
+                case "SHA3-256":
+                    return (this.p1363Format? new ECDSASignature.SHA3_256inP1363Format() :
+                    new ECDSASignature.SHA3_256());
+                case "SHA3-384":
+                    return (this.p1363Format? new ECDSASignature.SHA3_384inP1363Format() :
+                    new ECDSASignature.SHA3_384());
+                case "SHA3-512":
+                    return (this.p1363Format? new ECDSASignature.SHA3_512inP1363Format() :
+                    new ECDSASignature.SHA3_512());
+                default:
+                    return null;
```

This should throw an exception (other than a `NullPointerException`):
```java
                    throw new ProviderException("Unexpected algorithm: " + mdAlgo);
```

---

Changed to suggestion.